compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -2063,6 +2066,9 @@ fn encode_metadata_impl(tcx: TyCtxt<'_>) -> EncodedMetadata {
     // Encode the rustc version string in a predictable location.
     rustc_version().encode(&mut ecx).unwrap();
 
+    // Encode the crate's svh hash in a predictalbe location for cargo.
+    crate_svh(tcx).encode(&mut ecx).unwrap();
```

Comment:
You don't do the equivalent in the decoder. Also can you move it to above `rustc_version().encode(&mut ecx).unwrap()`? Otherwise cargo won't be able to know the right offset.

Reply:
Agreed it's nice to have it before the rustc version as that way it's at a deterministic location now that we've fixed the size to 512bits.